File: src/main/scala/com/yogpc/qp/data/QuarryDataGenerator.java
package com.yogpc.qp.data;

import com.yogpc.qp.QuarryPlus;
import net.fabricmc.fabric.api.datagen.v1.DataGeneratorEntrypoint;
import net.fabricmc.fabric.api.datagen.v1.FabricDataGenerator;

@SuppressWarnings("unused")
public final class QuarryDataGenerator implements DataGeneratorEntrypoint {
    @Override
    public void onInitializeDataGenerator(FabricDataGenerator fabricDataGenerator) {
        try {
            QuarryPlus.LOGGER.info("Quarry Data Generator is called.");
            var clazz = Class.forName("com.yogpc.qp.data.Generator");
            var instance = (DataGeneratorEntrypoint) clazz.getConstructor().newInstance();
            instance.onInitializeDataGenerator(fabricDataGenerator);
        } catch (ReflectiveOperationException ignore) {
        }
    }
}

File: fabric/src/dataGen/scala/com/yogpc/qp/fabric/data/QuarryDataGenerator.scala
package com.yogpc.qp.fabric.data

import net.fabricmc.fabric.api.datagen.v1.{DataGeneratorEntrypoint, FabricDataGenerator}

final class QuarryDataGenerator extends DataGeneratorEntrypoint {
  override def onInitializeDataGenerator(fabricDataGenerator: FabricDataGenerator): Unit = {
    val pack = fabricDataGenerator.createPack()
    pack.addProvider((o, r) => new RecipeFabric(o, r))
  }
}
